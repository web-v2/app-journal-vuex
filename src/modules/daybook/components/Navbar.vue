<template>
    <div>
        <nav class="navbar bg-primary">
            <a href="" class="navbar-brand text-white">
                <img src="@/assets/logo.png" alt="Vue-logo" class="d-inline-block align-text-top mx-2" height="24">
                {{username}}
            </a>
            <div class="d-flex">
                <button class="btn btn-outline-info" @click="onLogout">
                    <i class="fa fa-sign-out-alt"></i>
                </button>
            </div>
        </nav>
    </div>
</template>

<script>
import useAuth from '@/modules/auth/composables/useAuth'
import { useRouter } from 'vue-router'

export default {
    setup() {
        const router = useRouter()
        const {username, logout} = useAuth()

        return {
            username, 
            onLogout: ()=>{
                router.push({name: 'login'})
                logout()
            } 
        }
    }
}
</script>
