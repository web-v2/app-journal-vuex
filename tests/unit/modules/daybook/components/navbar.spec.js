import { shallowMount } from "@vue/test-utils";

import Navbar from "@/modules/daybook/components/Navbar.vue";
import createVuexStore from "../../../mock-data/mock-store";

describe("Pruebas en el navbar component", () => {
  const store = createVuexStore({
    status: "authenticated", // 'authenticated','not-authenticated', 'authenticating'
    user: { name: "SamirV2", email: "svergara@mail.com" },
    idToken: "ABC-123",
    refreshToken: "XYZ-123",
  });
  beforeEach(() => jest.clearAllMocks());
  test("Debe de mostrar el componente correctamente", () => {
    const wrapper = shallowMount(Navbar, {
      global: {
        plugins: [store],
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
  test("click en logout, debe de cerrar sesiÃ³n y redireccionar", async () => {
    const wrapper = shallowMount(Navbar, {
      global: {
        plugins: [store],
      },
    });

    await wrapper.find("button").trigger("click");

    expect(wrapper.router.push).toHaveBeenCalledWith({ name: "login" });

    expect(store.state.auth).toEqual({
      user: null,
      status: "not-authenticated",
      idToken: null,
      refreshToken: null,
    });
  });
});
