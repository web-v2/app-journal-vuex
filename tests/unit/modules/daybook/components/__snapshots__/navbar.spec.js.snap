// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el navbar component Debe de mostrar el componente correctamente 1`] = `
<div>
  <nav class="navbar bg-primary"><a href="" class="navbar-brand text-white"><img src="" alt="Vue-logo" class="d-inline-block align-text-top mx-2" height="24"> SamirV2</a>
    <div class="d-flex"><button class="btn btn-outline-info"><i class="fa fa-sign-out-alt"></i></button></div>
  </nav>
</div>
`;
